api/provider/aws/load_balancer_delete.go

```diff
@@ -0,0 +1,91 @@
+package aws
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/elb"
+	"github.com/aws/aws-sdk-go/service/iam"
+)
+
+func (l *LoadBalancerProvider) Delete(loadBalancerID string) error {
+	fqLoadBalancerID := addLayer0Prefix(l.Config.Instance(), loadBalancerID)
+
+	if err := l.deleteLoadBalancer(fqLoadBalancerID); err != nil {
```

Comment:
question: when deleting resources like r1, r2 & r3, we only check if an r actually exists or not for r3 (for example). Is there an implicit logic being applied here?